### PR TITLE
Create a new documentation package for Helm Chart

### PR DESCRIPTION
**Pull request by mik-laj**
Part of: https://github.com/apache/airflow/issues/14303

I have extracted the documentation from README.md to a separate package to make it easier for us to publish it on the site. This will also facilitate our documentation development as we have several files and not just one file. Previously, contributors didn't always add documentation to avoid cluttering the README.md file, which was already getting very long.'

For now, I have limited the changes I made to the content to make sure everything was moved, but in the following PR, we should update the content a bit more, eg decide how we will build the "Parameters" section.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/helm-chart/keda.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Autoscaling with KEDA
+---------------------
+
+*This feature is still experimental.*
+
+KEDA stands for Kubernetes Event Driven Autoscaling.
+`KEDA <https://github.com/kedacore/keda>`__ is a custom controller that
+allows users to create custom bindings to the Kubernetes `Horizontal Pod
+Autoscaler <https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/>`__.
+We have built a scaler that allows users to create scalers based on
```
Comment (turbaszek) [inline docs/helm-chart/keda.rst]:
```suggestion
We have built a scalers that allows users to create scalers based on
```

Comment (turbaszek) [reply docs/helm-chart/keda.rst]: There are 3 scalers that can be used with Airflow and all build by Daniel or me. But I assume we should mention only PostgreSQL later on as Postgres is part fo chart.

**Review by turbaszek — APPROVED** [review]
One small comment but otherwise this looks amazing! Thanks @mik-laj 🐈

File: docs/helm-chart/keda.rst (hunk 1 above)
Comment (mik-laj) [reply docs/helm-chart/keda.rst]: we should describe the scaller for MySQL as well, but I would prefer to do it as a separate ticket. for now, I am changing the structure, but without changing the content.

Helm Chart supports PostgreSQL and MySQL.  in cluster postgres is only intended for testing.

**Review by ashb — COMMENTED** [review]
Love this idea!

File: docs/helm-chart/keda.rst (hunk 1 above)
Comment (ashb) [reply docs/helm-chart/keda.rst]: `a scaler` or `scallers` (but not `a scallers`)

File: docs/helm-chart/keda.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Autoscaling with KEDA
+---------------------
+
+*This feature is still experimental.*
+
+KEDA stands for Kubernetes Event Driven Autoscaling.
+`KEDA <https://github.com/kedacore/keda>`__ is a custom controller that
+allows users to create custom bindings to the Kubernetes `Horizontal Pod
+Autoscaler <https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/>`__.
+We have built a scalers that allows users to create scalers based on
```
Comment (mik-laj) [inline docs/helm-chart/keda.rst]:
```suggestion
We have built scalers that allows users to create scalers based on
```

**Pull request merged**